pkgcore/operations/format.py

```diff
@@ -57,7 +58,9 @@ def fetch_one(self, fetchable, observer):
     def failed_fetch(self, fetchable, observer):
         # run pkg_nofetch phase for fetch restricted pkgs
         if 'fetch' in self.pkg.restrict:
-            build_ops = self.domain.build_pkg(self.pkg, observer)
```

Comment:
I'm not sure this line will work. As it passes raw pkg to build_pkg, it will not generate needed build operation from src_operation(build_operations). Seems like it needs PackageWrapper, so that it can call binded _generate_pkg_operations from _ConfiguredTree instead, which gets buildable and run actual nofetch on it.